plugin/evm/vm_test.go

```diff
@@ -2110,3 +2110,90 @@ func TestCreateHandlers(t *testing.T) {
 		require.ErrorIs(t, elem.Error, rpc.ErrMissingBatchResponse)
 	}
 }
+
+func TestGraniteDeactivatesBlockGasCost(t *testing.T) {
+	tests := []struct {
+		fork                 upgradetest.Fork
+		expectedError        error
+		expectedBlockGasCost *big.Int
+	}{
+		{
+			fork:          upgradetest.Fortuna,
+			expectedError: dummy.ErrInsufficientBlockGas,
+		},
+		{
+			fork:                 upgradetest.Granite,
+			expectedError:        nil,
+			expectedBlockGasCost: big.NewInt(0),
+		},
+	}
+
+	for _, test := range tests {
+		vm := newDefaultTestVM()
+		_ = vmtest.SetupTestVM(t, vm, vmtest.TestVMConfig{
+			Fork: &test.fork,
+		})
+
+		defer vm.Shutdown(context.Background())
+
+		vm.clock.Set(vm.clock.Time().Add(time.Second * 1))
+		tx := types.NewTransaction(uint64(0), vmtest.TestEthAddrs[1], big.NewInt(1), 21000, common.Big1, nil)
+		signedTx, err := types.SignTx(tx, types.NewEIP155Signer(vm.chainConfig.ChainID), vmtest.TestKeys[0].ToECDSA())
+		require.NoError(t, err)
+		blk, err := vmtest.IssueTxsAndSetPreference([]*types.Transaction{signedTx}, vm)
+		require.NoError(t, err)
+		require.NoError(t, blk.Accept(context.Background()))
+
+		vm.clock.Set(vm.clock.Time().Add(time.Second * 1))
+		// have another tx very quickly after the first one
+		tx = types.NewTransaction(uint64(1), vmtest.TestEthAddrs[1], big.NewInt(1), 21000, common.Big1, nil)
```

Comment:
Confirming my understanding...the reason why this `tx` has 0 priority fee per gas is because it is a legacy tx that sets the minimum possible (1 wei) gas price, right? 

It's the same exact functionally, but it may be more clear to do:
```
tx = types.NewTx(&types.DynamicFeeTx{
			ChainID: vm.chainConfig.ChainID,
			Nonce:   1,
			To:      &vmtest.TestEthAddrs[1],
			Value:   big.NewInt(1),
			Gas:     21000,
			GasFeeCap: common.Big1,
			GasTipCap: common.Big0,
		})
```
so that it's very explicit that there is no priority fee.

(Looks like `types.NewTransaction` is deprecated also either way)

Reply:
yea it was just a copy pasta from other tests.